brownie/project/compiler/__init__.py

```diff
@@ -270,91 +287,97 @@ def generate_build_json(
 
     Returns: build json dict"""
 
-    if input_json["language"] not in ("Solidity", "Vyper"):
-        raise UnsupportedLanguage(f"{input_json['language']}")
+    language: str = input_json["language"]
+    if language not in ("Solidity", "Vyper"):
+        raise UnsupportedLanguage(language)
 
     if not silent:
         print("Generating build data...")
 
     if compiler_data is None:
         compiler_data = {}
-    compiler_data["evm_version"] = input_json["settings"]["evmVersion"]
+
+    settings = input_json["settings"]
+    compiler_data["evm_version"] = settings["evmVersion"]
     build_json: Dict = {}
 
-    if input_json["language"] == "Solidity":
-        compiler_data["optimizer"] = input_json["settings"]["optimizer"]
+    if language == "Solidity":
+        compiler_data["optimizer"] = settings["optimizer"]
         source_nodes, statement_nodes, branch_nodes = solidity._get_nodes(output_json)
 
-    for path_str, contract_name in [
-        (k, x) for k, v in output_json["contracts"].items() for x in v.keys()
-    ]:
-        contract_alias = contract_name
-
-        if path_str in input_json["sources"]:
-            source = input_json["sources"][path_str]["content"]
+    sources: dict = input_json["sources"]
+    contracts: Dict[str, Dict[str, dict]] = output_json["contracts"]
+    for path_str, path_contracts in contracts.items():
+        if path_str in sources:
+            source = sources[path_str]["content"]
+            get_alias = False
         else:
             with Path(path_str).open(encoding="utf-8") as fp:
                 source = fp.read()
-            contract_alias = _get_alias(contract_name, path_str)
+            get_alias = True
 
-        if not silent:
-            print(f" - {contract_alias}")
+        for contract_name, contract in path_contracts.items():
+            contract_alias = _get_alias(contract_name, path_str) if get_alias else contract_name
 
-        abi = output_json["contracts"][path_str][contract_name]["abi"]
-        natspec = merge_natspec(
-            output_json["contracts"][path_str][contract_name].get("devdoc", {}),
-            output_json["contracts"][path_str][contract_name].get("userdoc", {}),
-        )
-        output_evm = output_json["contracts"][path_str][contract_name]["evm"]
-        if contract_alias in build_json and not output_evm["deployedBytecode"]["object"]:
-            continue
+            if not silent:
+                print(f" - {contract_alias}")
 
-        if input_json["language"] == "Solidity":
```

Comment:
@nvy-0x I dont want this whole section but try to do just the part that involves `language` like you did in the other parts